settings/config.go

```diff
@@ -36,31 +37,50 @@ func init() {
 	err := viper.ReadInConfig()
 
 	if err != nil {
-		image, pkgmanager, err := GetHostInfo()
+		panic("Failed to read the config file")
+	}
+
+	err = viper.Unmarshal(&Cnf)
+
+	if err != nil {
+		log.Fatalf("Config error: %v\n" + err.Error())
+	}
+}
+
+func GetDistroFromSettings() DistroInfo {
+	if Cnf.DistroId == "" || Cnf.Image == "" || Cnf.PkgManager == "" {
+		distro, err := GetHostInfo()
 
 		if err != nil {
 			log.Fatal("Unsupported setup detected, set a default distro and package manager in the config file")
 		}
 
-		Cnf = &Config{ContainerName: "apx_managed", Image: image, PkgManager: pkgmanager, DistroboxPath: "/usr/lib/apx/distrobox"}
+		Cnf = &Config{ContainerName: "apx_managed", Image: distro.Image, PkgManager: string(distro.Pkgmanager), DistroboxPath: "/usr/lib/apx/distrobox"}
```

Comment:
I think that the ContainerName should be set based on the PkgManager here, since all default names are set in `distros.go`

Reply:
> since all default names are set in `distros.go`

You should see them separate from here. This piece of code is reached when the "default" container is not set in the config file and it tries to autodetect the host. The containers in `distros.go` are purely for the `--<distro>` switches.

I disagree that they should be set based on the PkgManager. Like I said before in this thread the interesting part is not what package manager is used but the distribution. You could have both a Debian and a Ubuntu container (very feasible on e.g. a Debian host) and you wouldn't want them to conflict purely because they use the same package manager.